lisa/sut_orchestrator/azure/tools.py

```diff
@@ -55,3 +57,53 @@ def get_generation(self) -> str:
         else:
             generation = "1"
         return generation
+
+
+class LisDriver(Tool):
+    """
+    This is a virtual tool to detect/install LIS drivers.
+    """
+
+    __version_pattern = re.compile(r"^version:[ \t]*([^ \n]*)")
+
+    @property
+    def command(self) -> str:
+        return "modinfo hv_vmbus"
+
+    def _check_exists(self) -> bool:
+        if not self.node.os.is_posix:
+            return False
+
+        if isinstance(self.node.os, Fedora):
+            # currently LIS is only supported with Fedora
+            # and its derived distros
+            if self.node.os.package_exists(
+                "kmod-microsoft-hyper-v"
+            ) and self.node.os.package_exists("microsoft-hyper-v"):
+                return True
+        return False
+
+    def get_version(self) -> str:
+        cmd_result = self.run()
+        if cmd_result.exit_code == 0:
+            found_version = find_patterns_in_lines(
+                cmd_result.stdout, [self.__version_pattern]
+            )
+        return found_version[0][0] if found_version[0] else ""
+
+    def install_driver(self) -> None:
+        wget_tool = self.node.tools[Wget]
+        lis_path = wget_tool.get("https://aka.ms/lis", str(self.node.working_path))
+
+        result = self.node.execute(f"tar -xvzf {lis_path}")
+        if result.exit_code != 0:
+            raise LisaException(
+                "Failed to extract tar file after downloading LIS package."
+                f"stdout: {result.stdout}"
+            )
+        lis_folder_path = self.node.working_path.joinpath("LISISO")
+        result = self.node.execute("./install.sh", cwd=lis_folder_path)
+        if result.exit_code != 0:
+            raise LisaException(
+                "Unable to install the LIS RPMs!" f"stderr: {result.stderr}"
```

Comment:
1. log both exit_code and stderr, because the check condition is exit_code. 
2. If a string can be written in one line, it doesn't need to be split to two like this.